docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,544 @@
+  * We need to check the signature and length of the bitfield for validity. 
+  * We will keep the most recent bitfield for each validator in the session. Each bit corresponds to a particular parachain candidate pending availability. Parachains are scheduled on every block, so we can assign a bit to each one of those. Parathreads are not scheduled on every block, and there may be a lot of them, so we probably don't want a dedicated bit in the bitfield for those. Since we want an upper bound on the number of parathreads we have scheduled or pending availability, a concept of "execution cores" used in scheduling (TODO) should be reusable here - have a dedicated bit in the bitfield for each core, and each core will be assigned to a different parathread over time.
+  * Bits that are set to `true` denote candidate pending availability which are believed by this validator to be available.
+  * Candidates that are pending availability and have the corresponding bit set in 2/3 of validators' bitfields (only counting those submitted after the candidate was included, since some validators may not have submitted bitfields in some time) are considered available and are then moved into the "pending approval" state.
+  * Candidates that have just become available should apply any pending code upgrades based on the relay-parent they are targeting and should schedule any upcoming pending code upgrades.
+
+candidates entering the "pending approval" state:
+  * Apply fees (TODO: not sure if fees are actually used, we don't seem to need 'em for XCMP)
+  * Apply pending code upgrade, if any.
+  * Schedule a new pending code upgrade if the candidate specifies any. (there is a race condition here: part of the configuration is "how long should it take before pending code changes are applied". This value is computed based on the relay-parent that was used at the point when the candidate was about to be included in the relay chain. This is potentially a few blocks later than that, as it can take some time for a candidate to become fully available. We need to ensure that the code upgrade is scheduled with the same delay as was expected when the code upgrade was signaled. The easiest thing to do is to make sure the `pending_code_delay` is passed through the entire availability pipeline).
+  * Schedule Upwards messages - messages from the parachain to the relay chain.
+
+process new backed candidates:
+  * ensure that only one candidate is backed for each parachain or parathread
+  * ensure that the parachain or parathread of the candidate was scheduled and does not currently have a block pending availability. 
+  * check the backing of the candidate. 
+  * move to "pending approval" state. (pass along any configuration information that is liable to change)
+
+misbehavior reports and secondary checks:
+  * Secondary checks will also be submitted within the block. This may lead to slashing as a secondary check period ends.
```

Comment:
```suggestion
  * Secondary checks will also be submitted within the block. This may lead to slashing as a secondary check period ends. We want to catch and punish for the cases of misbehavior that violate the protocol and put it's security at risk. One of such cases is submitting conflicting votes on the same `CandidateReceipt`.Other examples include violations to AnV protocol or equivocations in finality. Misbehavior handling is implemented in
    * Runtime as an entry point.
    * Code in the Node that assists submitting misbehavior reports.
```